src/InfluxDB.LineProtocol/LineProtocolWriter.cs

```diff
@@ -177,29 +230,30 @@ public void Timestamp(long value)
                     throw InvalidPositionException("Cannot write timestamp as no field written for current measurement.");
             }
 
-            textWriter.Write(value.ToString(CultureInfo.InvariantCulture));
+            var timestamp = nanoseconds / (long)Precision;
+            textWriter.Write(timestamp.ToString(CultureInfo.InvariantCulture));
 
             position = LinePosition.TimestampWritten;
         }
 
-        public void Timestamp(TimeSpan value)
+        public void Timestamp(TimeSpan value, PrecisionResolutionStrategy? resolutionStrategy = null)
```

Comment:
These ones will be binary breaking changes - not a big deal, but we should either:

 * add overloads instead of default args, or
 * bump the major version to 2.0

Reply:
I was thinking it wasn't a breaking change as the new param is optional but on further investigation it appears it would be a breaking change as although it would work if recompiled it would break if ref updated without recompiling.

I will add overloads.